binary_search/binary_search.c

```diff
@@ -5,6 +5,39 @@
  * @arr_size Size of array
  * @search_element Element to be searched
  */
+ 
+/* function to perform recursive binary search
+ * input parameters:
+ * @ar - user-input array.
+ * @l - left index of array.
+ * @r - right index of array.
+ * @ele - element to be searched.
+ */
+int recursive_binary_search(const int *ar, int l, int r, int ele) {
```

Comment:
Also, every variable should have their own declaration statement. Please split these.